addons/mail/static/src/components/thread_icon/thread_icon.xml

```diff
@@ -5,14 +5,11 @@
         <t t-if="thread">
             <div class="o_ThreadIcon d-flex justify-content-center flex-shrink-0" t-attf-class="{{ className }}" t-ref="root" name="root">
                 <t t-if="thread.channel and thread.channel.channel_type === 'channel'">
-                    <t t-if="thread.public === 'private'">
-                        <div class="o_ThreadIcon_channelPrivate fa fa-fw fa-lock" title="Private channel"/>
+                    <t t-if="thread.authorizedGroupFullName">
+                        <div class="o_ThreadIcon_groupRestrictedChannel fa fa-fw fa-hashtag" t-attf-title="Selected Group of {{thread.authorizedGroupFullName}}"/>
```

Comment:
title text should be computed, to use translate function

Also the phrase should be something similar than `accessRestrictedToGroupText` (maybe the same)